File: iSeeCars.Business/Managers/CarManager.cs
using iSeeCars.Core.Entities;

namespace iSeeCars.Business.Managers
{
    public class CarManager
    {
        private readonly ModelManager modelManager;
        public CarManager(ModelManager modelManager) {
            this.modelManager = modelManager;
        }
        List<Car> cars = new List<Car>();
        private int nextId = 1;

        public void AddCar(Car car)
        {
            if (car == null)
                throw new ArgumentNullException();

            if (!modelManager.GetAllModels()
                .Any(x => x.ModelId == car.ModelId))
                throw new Exception("Model not found");

            car.CarId = nextId++;

            cars.Add(car);

        }
        #region ManualGetByid
        //public void GetCarById(int id)
        //{
        //    Car car = null;
        //    foreach (var item in cars)
        //    {
        //        if(id == item.CarId)
        //        {
        //            car = item;
        //            Console.WriteLine($"model name:{car.ModelId} price:{car.Price}, color:{car.Color}, fuel type:{car.FuelType}, engine: {car.Engine}, year:{car.Year.ToString("yyyy")}");
        //            break;
        //        }
        //    }

        //}
        #endregion

        public Car GetCarById(int id)
        {
            return cars.FirstOrDefault(c => c.CarId == id);
        }
        // Konsol üçün
        public void PrintCarById(int id, ModelManager modelManager)
        {
            var car = cars.FirstOrDefault(c => c.CarId == id);
            if (car == null)
            {
                Console.WriteLine("Car not found");
                return;
            }

            var model = modelManager.GetModelById(car.ModelId);
            string modelName = model?.ModelName ?? "Unknown";
            Console.ForegroundColor = ConsoleColor.Magenta;
            Console.WriteLine($"Model name:{modelName}, price:{car.Price}, color:{car.Color}, fuel type:{car.FuelType}, engine: {car.Engine}, year:{car.Year:yyyy}");
        }
        public List<Car> GetAllCars()
        {
            return cars;
        }
        public void ShowAllCars(ModelManager modelManager)
        {
            if (cars.Count == 0)
            {
                Console.WriteLine("Car not found");
            }
            else
            {

                foreach (var car in cars)
                {
                    var model = modelManager.GetModelById(car.ModelId);
                    string modelName = model.ModelName;
                    Console.ForegroundColor = ConsoleColor.Magenta;
                    Console.WriteLine($"id:{car.CarId},model name:{modelName}, price:{car.Price}, color:{car.Color}, fuel type:{car.FuelType}, engine: {car.Engine}, year:{car.Year.ToString("yyyy")}");
                }

            }
        }
        public void RemoveCar(Car car)
        {
            cars.Remove(car);
        }
    }
}

File: iSeeCars.java/src/main/java/iseecars/manager/CarManager.java
package iseecars.manager;
import static java.lang.System.out;
import iseecars.model.Car;

import java.util.ArrayList;
import java.util.List;

public class CarManager {
private List<Car> cars = new ArrayList<Car>();
private int nextId = 1;
    private ModelManager modelManager;

    public CarManager(ModelManager modelManager){

        this.modelManager = modelManager;
    }
    public void addCar(Car car){
        if(car == null){
            throw new IllegalArgumentException("Car cannot be null");
        }

        boolean modelExists = modelManager.getAllModels()
                .stream()
                .anyMatch(x -> x.getModelId() == car.getModelId());

        if(!modelExists){
            throw new RuntimeException("Model not found");
        }

        car.setCarId(nextId++);
        cars.add(car);
    }

    public Car getCarById(int id){
        return cars.stream()
                .filter(c -> c.getCarId() == id)
                .findFirst()
                .orElse(null);
    }

    public void printCarById(int id){
        Car car = getCarById(id);

        if(car == null){
            out.println("Car not found");
            return;
        }

        var model = modelManager.getModelById(car.getModelId());
        String modelName = (model != null) ? model.getModelName() : "Unknown";

        out.println(
                "Model name: " + modelName +
                        ", price: " + car.getPrice() +
                        ", color: " + car.getColor() +
                        ", fuel type: " + car.getFuelType() +
                        ", engine: " + car.getEngine() +
                        ", year: " + car.getYear()
        );
    }

    public List<Car> getAllCars(){
        return cars;
    }

    public void showAllCars(){
        if(cars.isEmpty()){
            out.println("Car not found");
            return;
        }


    for(Car car : cars){
        var model = modelManager.getModelById(car.getModelId());
        String modelName = model != null ? model.getModelName() : "Unknown";

        out.println(
                "id: " + car.getCarId() +
                        ", model name: " + modelName +
                        ", price: " + car.getPrice() +
                        ", color: " + car.getColor() +
                        ", fuel type: " + car.getFuelType() +
                        ", engine: " + car.getEngine() +
                        ", year: " + car.getYear()
        );
    }
    }
    public void removeCar(Car car){
        cars.remove(car);
    }
}
